src/common

```diff
@@ -101,17 +101,19 @@ else ifeq ($(BUILD_ENV_),Cygwin)
   CXXFLAGS ?= -Wall -O2
   CP_CFLAGS = -MMD -MP
   CP_CXXFLAGS += -MMD -MP -std=c++11 -U__STRICT_ANSI__
+  # Cygwin-g++ has problems with statically linking exception code.
+  # If linking fails, remove -static.
   LINKFLAGS = -static -std=c++11
   LINKLIB = ar rcT $@ $^
-  LINKBIN = $(CXX) $(LINKFLAGS) -o $@ -Wl,--start-group $^ -Wl,--end-group $(LIBS) -static
+  LINKBIN = $(CXX) $(LINKFLAGS) -o $@ -Wl,--start-group $^ -Wl,--end-group $(LIBS)
   LINKNATIVE = $(HOSTCXX) -std=c++11 -o $@ $^ -static
 ifeq ($(origin CC),default)
   #CC = gcc
```

Comment:
32 bit compilers (more precisely compilers where sizeof(size_t) == 32) will not work because of justified errors like this:
```
interpreter.cpp: In member function ‘size_t interpretert::get_size(const typet&):
interpreter.cpp:971:20: error: large integer implicitly truncated to unsigned type [-Werror=overflow]
return 2ULL << 32ULL;
```